client/src/pages/Edit.js

```diff
@@ -17,17 +18,17 @@ function Edit() {
 
   useEffect(() => {
     const editFoodId = async () => {
-      queryAPI({
+      const foodResult = await queryAPI({
         url: `http://localhost:5000/menu_items/${id}`,
-        //more functionality with onSuccess Callback
-        onSuccess: (result) => toast.success(result.data, { timeout: 2000 }),
         onError: (result) => toast.error(result.error),
         method: "GET",
       });
+      console.log(data);
+      // console.log(foodResult);
+      dataSet(foodResult);
```

Comment:
I see this approach caused you to change your value props on the inputs below to `data.data.title`, `data.data.author` and `data.data.img`.  If you want, you can *spread* the value of `foodResult` here and return those props to their previous implementation.  

```suggestion
      dataSet(...foodResult);
```